File: web-common/src/lib/number-formatting/utils/count-digits.ts
export const countDigits = (numStr: string) => {
  if (numStr === "0") return 0;
  return numStr.replace(/[^0-9]/g, "").length;
};

export const countNonZeroDigits = (numStr: string) =>
  numStr.replace(/[^1-9]/g, "").length;

File: runtime/pkg/formatter/count_digits.go
package formatter

import "regexp"

var (
	digits        = regexp.MustCompile(`\D`)
	nonZeroDigits = regexp.MustCompile(`[^1-9]`)
)

// countDigits counts all numeric digits in a string.
func countDigits(numStr string) int {
	return len(digits.ReplaceAllString(numStr, ""))
}

// countNonZeroDigits counts all non-zero numeric digits in a string.
func countNonZeroDigits(numStr string) int {
	return len(nonZeroDigits.ReplaceAllString(numStr, ""))
}
